### PR TITLE
[SPARK-27540][MLlib] Add 'meanAveragePrecision_at_k' metric to RankingMetrics

### PR DESCRIPTION
**Pull request by pntuananh**
## What changes were proposed in this pull request?

Add 'meanAveragePrecision_at_k' metric to RankingMetrics

## How was this patch tested?

Add test to RankingMetricsSuite.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala [hunk 1]
```diff
@@ -70,26 +70,42 @@ class RankingMetrics[T: ClassTag](predictionAndLabels: RDD[(Array[T], Array[T])]
...
+      var i = 0
+      var cnt = 0
+      var precSum = 0.0
+      val n = math.min(pred.length, k)
+      while (i < n) {
+        if (labSet.contains(pred(i))) {
+          cnt += 1
+          precSum += cnt.toDouble / (i + 1)
         }
-        precSum / labSet.size
```
Comment (qb-tarushg) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala]: I think division would be changed to below :
```scala
precSum / math.min(labSet.size, k)
```

Comment (pntuananh) [reply mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala]: @qb-tarushg : based on definition of Average Precision, we should divide by number of relevant items, not by k.

**Comment by pntuananh** [comment]
Can someone verify this?

**Comment by srowen** [comment]
How is this related to https://github.com/apache/spark/pull/24458 ? let's have only one PR

**Comment by holdenk** [comment]
Jenkins OK to test

**Pull request closed without merging**